### PR TITLE
feat: use changelog.rst instead of CHANGELOG.md for release notes

### PR DESCRIPTION
**Pull request by klmcadams**
Use `changelog.rst` file to create a "Release notes" tab in the documentation instead of using `CHANGELOG.md` to record release notes

### CODE REVIEW

File: doc/source/changelog.rst [hunk 1]
```diff
@@ -0,0 +1,922 @@
+.. _ref_release_notes:
+
+Release notes
+#############
+
+This document contains the release notes for the project.
+
+.. vale off
```
Comment (dipinknair) [inline doc/source/changelog.rst:8]: @klmcadams  should we turn on vale at the end?

**Review by dipinknair — APPROVED** [review]
LGTM !

File: doc/source/changelog.rst (hunk 1 above)
Comment (klmcadams) [reply doc/source/changelog.rst:8]: Ohh yeah, it should be right after the towncrier release notes start line. I will add it

Comment (klmcadams) [reply doc/source/changelog.rst:8]: You're right, it should be at the end. I was referring to the setup guide oops

**Pull request merged**